blog/2021-05-19-monero-transaction.md

```diff
@@ -0,0 +1,280 @@
+Nonetheless, we would still like to mention all the particularities of Monero's implementation of CLSAG.
+
+###### Involving Pedersen commitments
+
+Vanilla CLSAG (as described in _Zero to Monero_) only concerns itself with authorising spending an input, but Monero's implementation has been modified to also confidentially prove that the real input used commits to the same amount as the pseudo output commitment included in the transaction.
+
+An iteration of the CLSAG algorithm commonly consists of computing a challenge of the form `c_i+1 = H_n(domain_tag || public_parameters || L_i+1 || R_i+1)`, where the domain tag and the public parameters remain constant between iterations, and `L` and `R` are expressions which depend on the previous iteration.
+We can therefore focus on the values of `L` and `R` and see how they're computed differently in Monero.
+
+Looking at `L`, regular CLSAG computes it as `L_i+1 = r_i * G + c_i * pk_i ` whereas Monero's instead does `L_i+1 = r_i * G + c_i * mu_P * pk_i + c_i * mu_C * (C_i - C_pseudo)`, where `mu_P` and `mu_C` are constants, `C_i` is the `i`th commitment in the ring, `r_i` is the `i`th response and `C_pseudo` is the pseudo output commitment.
+We can see that a third component has been introduced in Monero, which is kept orthogonal to the second one because `mu_P ≠ mu_C`.
+This new component is actually responsible for attesting the fact that the pseudo output commitment hides the same amount as the real input's commitment.
+
+Normally CLSAG enforces that to be able to _close_ the ring and generate the response `r_π` we need to know the real input's secret key `sk_π`: `r_π = ⍺ - c_π * sk_π`, where `⍺` is a random value generated at the start of the algorithm.
+CLSAG _seeds_ the algorithm by first generating `c_π+1` based on `⍺`.
+More specifically, `L_π+1 = ⍺ * G`, which is successfully closed by `r_π`:
+first rearranging to get `⍺ = r_π + c_π * sk_π`; and then substituting in `L_π+1` giving `L_π+1 = r_π * G + c_π * sk_π * G = r_π * G + c_π * pk_π * G`, exactly what we'd expect from the definition.
+
+In Monero's CLSAG we instead calculate `r_π = ⍺ - c_π * mu_P * sk_π - c_π * mu_C * z`, where `z` is the difference between the real commitment's blinding factor and the pseudo output's commitment blinding factor.
+If we rearrange as above we get `⍺ = r_π + c_π * mu_P * sk_π + c_π * mu_C * z`, which when substituted in `L_π+1` gives `L_π+1 = r_π * G + c_π * mu_P * sk_π * G + c_π * mu_C * z * G` which is equal to `L_π+1 = r_π * G + c_π * mu_P * pk_π + c_π * mu_C * (C_π - C_pseudo)` if and only if `z * G = C_π - C_pseudo`.
+Since the real input's commitment and the pseudo output's commitment hide the same amount `C_π - C_pseudo = z_0 * G + a * H - (z_1 * G + a * H) = (z_0 - z_1) * G`.
+Therefore we can simplify our condition to `z * G = (z_0 - z_1) * G`, which is true given our definition of `z` above.
```

Comment:
Very nice! Things finally make sense on why we need the commitment ring during the signature generation :)